### PR TITLE
load: allow "global" config

### PR DESCRIPTION
**Pull request by yields**
This allows users to add "global" flags, for example:

```golang
  conf (gs) go run example/subcommand.go -baz foo cmd -foo foo
hello from `cmd`, root_conf: {Baz:foo} cmd_conf: {Foo:foo}
```

### CODE REVIEW

File: load.go [hunk 1]
```diff
@@ -110,35 +110,36 @@ func (ld Loader) Load(cfg interface{}) (cmd string, args []string, err error) {
...
 			return
 		}
 
 		found := false
 		for _, c := range ld.Commands {
-			if c.Name == ld.Args[0] {
-				found, cmd, ld.Args = true, ld.Args[0], ld.Args[1:]
+			if c.Name == args[0] {
+				found, cmd = true, args[0]
+				args = args[1:]
```
Comment (achille-roussel) [inline load.go]: Weird style here, either put all assignments on the same line or all on different lines:
```go
found, cmd, args = true, args[0], args[1:]
```
```go
found = true
cmd = args[0]
args = args[1:]
```

**Comment by yields** [comment]
> Do you think it's OK to have things like this be both valid?

hmm, i'm not sure, i personally like clear ordering, the ordering matters with this implementation -- that's because parent commands will strip away arguments as they parse them.


> I believe this is what this PR enables

This PR simply allows global config and strips the args, without this PR, global config doesn't work.

**Comment by achille-roussel** [comment]
Alright we discussed *offline*, I was confused about the behavior, this looks great 👍

**Comment by yields** [comment]
thanks!

**Comment by achille-roussel** [comment]
@yields I think this broke a bunch of cases when you have subcommands. For example:
```
command subcommand -h
```
should print the help of the subcommand but it doesn't anymore because it's caught as the help of the main command.

Do you min if we revert?

**Comment by yields** [comment]
> Do you min if we revert?

no, feel free to revert.

sorry about this, i thought it'd be alright because tests passed.

**Comment by achille-roussel** [comment]
I guess I could write better tests 😛

**Comment by yields** [comment]
> I guess I could write better tests 😛

or you could force PRs to include tests haha :)

**Pull request merged**